Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/shared/make-private.tsx

```diff
@@ -11,20 +11,33 @@ import { useAuth } from '../../hooks/use-auth';
 type IPrivatePageProps = IHaveChildrenProps
 
 const PrivatePage = ({ children }: IPrivatePageProps) => {
-    const { state: { user } } = useAuth();
+    const {
+        state: {
+            isLoggedIn,
+            isGetAuthInfoUnauthorized,
+            hasCompletedGetAuthInfo,
+        },
+    } = useAuth();
     const location = useLocation();
-    const { isLoggedIn } = user;
 
-    const state = { from: location };
+    const render = useCallback(() => {
+        // Do not render if has not attempted to load user or not logged in
+        if ((!hasCompletedGetAuthInfo || !isGetAuthInfoUnauthorized) && !isLoggedIn) {
+            // User still not loaded in state
+            return null;
```

Comment:
We could display some text like "Loading user data" while fetching the user data.

Reply:
Well, this flashes for a split second and you can barely see it, but i'll add it anyway.